modules/katello_sync.py

```diff
@@ -107,115 +104,57 @@
-        repository.organization = product.organization
-        response = repository.search()
+    from ansible.module_utils.ansible_nailgun_cement import (
+        find_organization,
+        find_product,
+        find_repository,
+    )
 
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Repository found for %s" % name)
+    from nailgun import entity_mixins
+    entity_mixins.TASK_TIMEOUT = 180000  # Publishes can sometimes take a long, long time
+except ImportError:
+    pass
 
-    def sync(self, product, organization, synchronous, repository=None):
-        if repository is None:
-            product = self.find_product(product, organization)
-            return product.sync(synchronous)
-        else:
-            repository = self.find_repository(repository, product, organization)
-            return repository.sync(synchronous)
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule
 
 
 def main():
-    module = AnsibleModule(
+    module = ForemanAnsibleModule(
         argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True, no_log=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
-            product=dict(required=True),
-            synchronous=dict(type='bool', default=True),
             organization=dict(required=True),
+            product=dict(required=True),
             repository=dict(),
+            synchronous=dict(type='bool', default=True),
         ),
         supports_check_mode=False,
     )
 
-    if has_import_error:
-        module.fail_json(msg=import_error_msg)
-
-    server_url = module.params['server_url']
-    verify_ssl = module.params['verify_ssl']
-    username = module.params['username']
-    password = module.params['password']
-    repository = module.params['repository']
-    product = module.params['product']
-    synchronous = module.params['synchronous']
-    organization = module.params['organization']
-
-    server = ServerConfig(
-        url=server_url,
-        auth=(username, password),
-        verify=verify_ssl
-    )
-    ng = NailGun(server, entities, module)
+    params = module.parse_params()
+
+    module.connect()
 
-    # Lets make an connection to the server with username and password
     try:
-        org = entities.Organization(server)
-        org.search()
+        organization = find_organization(module, params['organization'])
+        product = find_product(module, params['product'], organization)
+        if 'repository' in params:
+            repository = find_repository(module, params['repository'], product)
+        else:
+            repository = None
     except Exception as e:
-        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+        module.fail_json(msg='Failed to find entity: %s ' % e)
+
+    synchronous = params['synchronous']
```

Comment:
what's the benefit of having it here as a var, instead of passing `params['synchronous']` directly to the `sync()` calls in L155 and L157?

Reply:
Nothing, just one step less refactoring iirc.
I'm probably going to change that after having another look.